lib/integrations.rb

```diff
@@ -1,11 +1,16 @@
 # frozen_string_literal: true
 module Integrations
   class Error < StandardError
-    attr_reader :type, :message
+    attr_reader :type, :message, :response, :response_body, :response_code
 
-    def initialize(type='unknown_error', message='Unknown Error')
+    def initialize(type='unknown_error', message='Unknown Error', response)
       @type = type
       @message = message
+      if response && response.respond_to?(:body)
+        @response = response
+        @response_body = response.body
+        @response_code = response.code
```

Comment:
double checked (we're not handling hipchat failures with this new code, the hipchat gem seems to do a good job of doing that on it's own) and we seem to be good. the oauth response (for jira) and the httparty response (for slack and pivotal) objects both respond to `code` and `body`